e2e/README.md

```diff
@@ -1,92 +1,62 @@
-
-> In this test framework, we are running Godog with `go test`, as explained [here](https://github.com/cucumber/godog#running-godog-with-go-test).
-
-### Cucumber: BDD at its core
-
-The specification of these E2E tests has been done using `BDD` (Behaviour-Driven Development) principles, where:
-
->BDD aims to narrow the communication gaps between team members, foster better understanding of the customer and promote continuous communication with real world examples.
-
-From Cucumber's website:
-
->Cucumber is a tool that supports Behaviour-Driven Development(BDD), and it reads executable specifications written in plain text and validates that the software does what those specifications say. The specifications consists of multiple examples, or scenarios.
-
-The way we are going to specify our software is using [`Gherkin`](https://cucumber.io/docs/gherkin/reference/).
-
->Gherkin uses a set of special keywords to give structure and meaning to executable specifications. Each keyword is translated to many spoken languages. Most lines in a Gherkin document start with one of the keywords.
-
-The key part here is **executable specifications**: we will be able to automate the verification of the specifications and potentially get a coverage of these specs.
-
-### Godog: Cucumber for Go
+## Statements about the test framework
+- It uses the `Given/When/Then` pattern to define scenarios.
+  - `Given` a state is present (optional)
+  - `When` an action happens (mandatory) 
+  - `Then` an outcome is expected (mandatory)
+  - "And/But" clauses are allowed to provide continuation on the above ones. (Optional)
+- Because it uses BDD, it's possible to combine existing scenario steps (the given-when-then clauses) forming new scenarios, so with no code it's possible to have new tests. For that reason, the steps must be atomic and decoupled (as any piece of software: low coupling and high cohesion).
+- It does not use the GUI at all, so no Selenium, Cypress or any other test library having expectations on graphical components. It uses Go as the programming language and Cucumber to create the glue code between the code and the plain English scenarios. In almost two years, we have demonstrated that the APIs are not changing as fast as the GUI components.
+- APIs not changing does not mean zero flakiness. Because there are so many moving pieces (stack versions), beats versions, elastic-agent versions, cloud machines, network access, etc... There could be situations where the tests fail, but they are rarely caused by test flakiness. Instead, they are usually caused by: 1) instability of the all-together system, 2) a real bug, 3) a badly written test.
```

Comment:
```suggestion
- APIs not changing does not mean zero flakiness. Because there are so many moving pieces (stack versions), beats versions, elastic-agent versions, cloud machines, network access, etc... There could be situations where the tests fail, but they are rarely caused by test flakiness. Instead, they are usually caused by: 1) instability of the all-together system, 2) a real bug, 3) gherkin steps that are not consistent.
```